packages/card/index.test.js

```diff
@@ -0,0 +1,151 @@
+    const cardMeta = ['Category']
+
+    render(<Card {...defaultProps} meta={cardMeta} />)
+
+    const metaElement = screen.getByTestId('wpl-card-meta')
+    expect(metaElement).toContainElement(screen.getByText(cardMeta[0]))
+  })
+
+  it('should render the provided meta prop (two items)', () => {
+    const cardMeta = ['August 15, 2022', 'Category']
+
+    render(<Card {...defaultProps} />)
+
+    const metaElement = screen.getByTestId('wpl-card-meta')
+
+    cardMeta.forEach((item) => {
+      expect(metaElement).toContainElement(screen.getByText(item))
+    })
+    expect(metaElement).toContainElement(screen.getByText('|'))
+  })
+
+  it('should render the provided thumbnail prop', () => {
+    const { getByAltText } = render(<Card {...defaultProps} />)
+
+    expect(getByAltText(defaultProps.thumbnail.alt).tagName).toBe('IMG')
+  })
+
+  // Children render tests
+
+  it('should render provided Thumbnail sub-component children', () => {
+    const { getByAltText } = render(
+      <Card link={defaultProps.link}>
+        <Card.Thumbnail {...defaultProps.thumbnail} />
+      </Card>
+    )
+
+    expect(getByAltText(defaultProps.thumbnail.alt).tagName).toBe('IMG')
+  })
+
+  it('should render provided Meta sub-component children', () => {
+    render(
+      <Card link={defaultProps.link}>
+        <Card.Meta items={defaultProps.meta} />
+      </Card>
+    )
+
+    const metaElement = screen.getByTestId('wpl-card-meta')
+
+    defaultProps.meta.forEach((item) => {
+      expect(metaElement).toContainElement(screen.getByText(item))
+    })
+  })
+
+  it('should render provided Heading sub-component children', () => {
+    render(
+      <Card link={defaultProps.link}>
+        <Card.Heading>{defaultProps.heading}</Card.Heading>
+      </Card>
+    )
+
+    screen.getByText(defaultProps.heading)
+  })
+
+  it('should render provided Description sub-component children', () => {
+    render(
+      <Card link={defaultProps.link}>
+        <Card.Description>{defaultProps.description}</Card.Description>
+      </Card>
+    )
+
+    screen.getByText(defaultProps.description)
+  })
+
+  it('should render arbitrary JSX children', () => {
+    const arbitraryText = 'Arbitrary paragraph text'
+    const arbitraryParagraph = <p>{arbitraryText}</p>
+
+    render(<Card link={defaultProps.link}>{arbitraryParagraph}</Card>)
+
+    screen.getByText(arbitraryText)
+  })
+
+  it('should render arbitrary text content', () => {
+    const arbitraryText = 'Arbitrary text'
+
+    render(<Card link={defaultProps.link}>{arbitraryText}</Card>)
+
+    screen.getByText(arbitraryText)
+  })
+
+  it('should not render props-based layout if children are provided', () => {
+    const arbitraryText = 'Arbitrary text'
+
+    render(
+      <Card link={defaultProps.link} heading={defaultProps.heading}>
+        {arbitraryText}
+      </Card>
+    )
+
+    screen.getByText(arbitraryText)
+    expect(screen.queryByText(defaultProps.heading)).not.toBeInTheDocument()
```

Comment:
I tried using `getByText()` here and that always failed because the heading component is not supposed to be in the document, so it never was able to be found.

That's when I found `queryByText()`! The difference? `getByText()` will fail the test if it can't find the text, but `queryByText()` will return `null` instead. 

So, if `queryByText()` finds a heading element here, the test will fail as intended.